### PR TITLE
Update spark to 1.4.1 and client to 0.2.0

### PR DESCRIPTION
**Pull request by tnachen**
This includes upgrading spark to 1.4.1, and also let the client use spark 1.4.1 image as well.
There are also fixes that are applied to client and new features such as tailing driver log.

### CODE REVIEW

**Comment by kensipe** [comment]
@tnachen starting review to merge

File: repo/packages/S/spark/1/marathon.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+  "id": "spark",
```
Comment (kensipe) [inline repo/packages/S/spark/1/marathon.json]: we are looking for id, cpus, mem, instances to all have config.json properties so they can be modified at installation

Comment (tnachen) [inline repo/packages/S/spark/1/marathon.json]: Oh? even id? ok let me make this change

**Comment by kensipe** [comment]
@tnachen I rebased to the latest and I will update with the needed changes unless you are in the middle of it...

**Pull request merged**